client/state/form/reducer.js

```diff
@@ -14,6 +20,14 @@ reducers[ SET_FORM_PROPERTY ] = ( state, action ) => {
 	return Object.assign( {}, state, newObj );
 };
 
+reducers[ NEXT_STEP ] = ( state ) => {
+	return Object.assign( {}, state, { currentStep: state.currentStep + 1 } );
```

Comment:
Yep, this needs to be checked somewhere, but maybe it will be better in the action creator (conditional dispatch, one of the primary use-cases of `redux-thunk`). I still need to figure this out.